test/extended/router/router.go

```diff
@@ -50,9 +50,15 @@ var _ = g.Describe("[sig-network][Feature:Router][apigroup:operator.openshift.io
 	oc = exutil.NewCLIWithPodSecurityLevel("router-stress", admissionapi.LevelBaseline)
 
 	g.BeforeEach(func() {
-		var err error
-		host, err = exutil.WaitForRouterServiceIP(oc)
+		apiResourceExistsFn, err := exutil.DoesApiResourceExistCached(oc.AdminConfig())
 		o.Expect(err).NotTo(o.HaveOccurred())
+		if apiResourceExistsFn("ingresscontrollers") {
```

Comment:
I suppose we could now move this conditional logic inside the `WaitForRouterServiceIP`, as I am not seeing any bad implications for the other tests.

Do we want to move it there or limit the changes/scope of this PR?